canonical-web-and-design/maas-ui
# Build add chassis form.

## Done
- Added machine addChassis http method and saga/actions/reducers. LP bug about websocket implementation here: https://bugs.launchpad.net/maas/+bug/1863906
- Built AddChassisForm component.
- Refactored the AddMachineForm to use some common hooks and components.
- Note that chassis types are currently hardcoded, as in the legacy version. Relevant LP bug here: https://bugs.launchpad.net/maas/+bug/1863905

## QA
- Click the Add hardware dropdown and select Chassis.
- Check that you can change power type and the power parameters fields change.
- Check form validation works as expected.
- Enter a hostname/address that your MAAS is aware of (in my case I used karura's only KVM's virsh address) and submit.
- Check that a spinner shows and then you are redirected to the machine list with a confirmation notification.
- Enter incorrect data and submit and check that you get an error notification above the form.
- Check that you can still add machines in the Add machine form

## Fixes
Fixes canonical-web-and-design/MAAS-squad#1730
Fixes #817 

## Screenshots
### Default
![0 0 0 0_8400_MAAS_r_machines_chassis_add (1)](https://user-images.githubusercontent.com/25733845/74929366-a0445f00-53db-11ea-905b-711a348a66c5.png)


### Form validation
![0 0 0 0_8400_MAAS_r_machines_chassis_add (2)](https://user-images.githubusercontent.com/25733845/74929372-a4707c80-53db-11ea-9577-32c349b56138.png)


### Server error
![0 0 0 0_8400_MAAS_r_machines_chassis_add (3)](https://user-images.githubusercontent.com/25733845/74929379-a6d2d680-53db-11ea-9522-604f7eed0e1c.png)


### Confirmation notification
![0 0 0 0_8400_MAAS_r_machines_chassis_add (5)](https://user-images.githubusercontent.com/25733845/74929387-aafef400-53db-11ea-8e40-2cb9a96426b8.png)
